### PR TITLE
Bug fix and tests for any() Permission Evaluation

### LINKED ISSUE #189: PermissionDSL incorrect behaviour with ANY() permissions mapping

### PR DESCRIPTION
**Pull request by jmbrunskill**
Closes #189 

Added Tests for the PermissionDSL.

The PermissionsDSL allows you to specific combinations of permissions such as `PermissionDSL::Any(Permission::A,Permission::B)` which should mean either Permission::A or Permission::B should satisfy the permissions check.

Previously this was being evaluated incorrectly.

### CODE REVIEW

File: server/service/src/auth.rs [hunk 1]
```diff
@@ -357,7 +366,7 @@ fn validate_resource_permissions(
                     resource_request,
                     child,
                 ) {
-                    ()
+                    return Ok(());
```
Comment (jmbrunskill) [inline server/service/src/auth.rs:369]: This is the only code change, return the result as soon as any one permission is matched.

**Comment by jmbrunskill** [comment]
There's probably more test cases that could be created but I think these cover enough to be fairly confident.
Feel free to suggest more :)

**Review by clemens-msupply — APPROVED** [review]
Thanks!

nit: "//Test" -> "// Test" at least thats what we are usually using

**Review by andreievg — APPROVED** [review]
Had a really quick look, thanks.

**Comment by jmbrunskill** [comment]
> Thanks!
> 
> nit: "//Test" -> "// Test" at least thats what we are usually using

Will try to remember for next time :)

**Pull request merged**